src/components/PeriodicReports/Upload/useUploadPeriodicReport.ts

```diff
@@ -1,33 +1,45 @@
 import { useMutation } from '@apollo/client';
-import {
-  HandleUploadCompletedFunction,
-  useUploadFiles,
-} from '../../files/hooks';
+import { CalendarDate } from '../../../util';
+import { useUploadFiles } from '../../files/hooks';
 import { UploadPeriodicReportDocument } from './UploadPeriodicReport.generated';
 
-export const useUploadPeriodicReport = (reportId?: string) => {
+export const useUploadPeriodicReport = () => {
   const uploadFiles = useUploadFiles();
   const [uploadFile] = useMutation(UploadPeriodicReportDocument);
 
-  const handleUploadCompleted: HandleUploadCompletedFunction = async ({
-    uploadId,
-    name,
-    parentId: reportId,
-  }) => {
-    await uploadFile({
-      variables: {
-        input: {
-          reportId,
-          file: { uploadId, name },
+  return async (
+    reportId: string,
+    files?: File[],
+    receivedDate?: CalendarDate
+  ) => {
+    const id = reportId;
+    const updateReport = async (
+      reportId: string,
+      uploadId?: string,
+      name?: string
+    ) => {
+      await uploadFile({
+        variables: {
+          input: {
+            reportId,
+            ...(uploadId ? { file: { uploadId, name } } : {}),
+            receivedDate,
+          },
         },
-      },
-    });
-  };
-
-  return (files: File[], reportIdOverride?: string) => {
-    const id = reportIdOverride ?? reportId;
+      });
+    };
     if (id) {
```

Comment:
We needed to be able to update a report with just the received date, so I made the reportId param required since it can't be derived without a file input param. I'm not sure what the function of the report id override was and if we need to maintain that.